rama-dns/src/chain.rs

```diff
@@ -0,0 +1,192 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use rama_net::address::Domain;
+
+use crate::DnsResolver;
+use rama_core::combinators::Either;
+
+/// An error that occurs when a DNS resolver chain fails to resolve a domain.
+#[derive(Debug)]
+pub struct DnsChainDomainResolveErr<E: 'static> {
+    errors: Vec<E>,
+}
+
+impl<E: std::fmt::Debug> std::fmt::Display for DnsChainDomainResolveErr<E> {
```

Comment:
When you use BoxError internally this generic bound is no longer needed and won't exist